### DISCUSSION TITLE
JSON Objects as Inputs to Workflows

### DISCUSSION THREAD

**Discussion by quintenrosseel** [langgenius/dify#8903] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

I'm trying to use my Dify workflow as a tool call, from another service. (e.g. VAPI.ai). 

However, the Start block of Dify doesn't allow me to select a JSON `Object` as an input type. 
Is there a way so that I can call my Dify workflow with a JSON object as input? Other blocks clearly allow this as IO, so I'm wondering why I cannot use this. 

So, a POST request to `https://api.dify.ai/v1/workflows/run` with the following body: 
```json 
{
   "inputs": {
      "obj_attr": {
        "list": ["a", "b", "c,"], 
        "obj": {
           "nested_attribute": "Hello World!" 
         }
      }
   }, 
   "response_mode": "blocking"
   "user": "test-123"
}
```

And being able to parse that body as part of the workflow. 
The issue is that I cannot select a `Object` as input type. 

![image](https://github.com/user-attachments/assets/06989a27-c79c-44c1-bdfe-c10df5869e03)

### 3. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by perzeuss** [comment]
For now, please use paragraph and pass the JSON as string. You can parse the JSON with a code block.

**Reply by quintenrosseel** [reply]
Thank you for your suggestion. 🙌
I'll give this a try, however: 

1.  I don't always control how other tools perform tool calls. This shifts the problem upstream.
2. This feels hacky. 

So, a longer-term and consistent solution would be appreciated. 

Thanks!

**Reply by laelhalawani** [reply]
i agree this isn't a great solution and more of a hack than actual solution, objects should be generally supported as inputs at least to some extent

**Comment by Yevanchen** [comment]
We will support this capability soon.

**Reply by DavideDelbianco** [reply]
Please take into consideration also #19593 #21522 if object are going to be supported, boolean / arrays might as well be supported

**Reply by Yevanchen** [reply]
Arrays may not be supported

**Reply by AlexChim1231** [reply]
any progress? can the object contain dify file object?

**Reply by dheeraj-circuitry** [reply]
any progress. Thanks

**Reply by s1mbah** [reply]
any progress?
